crates/typst/src/layout/inline/mod.rs

```diff
@@ -1446,3 +1489,56 @@ fn overhang(c: char) -> f64 {
         _ => 0.0,
     }
 }
+
+/// Whether the hyphen should repeat at the start of the next line.
+fn should_repeat_hyphen(pred_line: &Line) -> bool {
+    // If the predecessor line does not end with a Dash::HardHyphen, we shall not place a hyphen at
+    // the start of the next line.
+    if pred_line.dash != Some(Dash::HardHyphen) {
+        return false;
+    }
+
+    // If there's a trimmed out space, we needn't repeat the hyphen. That's the case of a text like
```

Comment:
could you wrap these comments at 80 columns?

Reply:
Sure. Done at last commit.